pkg/build/controller/buildpod/controller.go

```diff
@@ -236,7 +246,7 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 		case buildapi.BuildPhaseError, buildapi.BuildPhaseFailed:
 			bc.recorder.Eventf(build, kapi.EventTypeNormal, buildapi.BuildFailedEventReason, fmt.Sprintf(buildapi.BuildFailedEventMessage, build.Namespace, build.Name))
 		}
-		common.HandleBuildCompletion(build, bc.runPolicies)
+		common.HandleBuildCompletion(build, bc.buildLister, bc.buildConfigGetter, bc.buildDeleter, bc.runPolicies)
```

Comment:
Handling this only when a running build transitions to a new phase is not optimal. For example, if I set the limits to a BC without running any build, I would expect any prunable builds to be removed.